tools/lsp/main.rs

```diff
@@ -313,12 +327,16 @@ pub fn run_lsp_server() -> Result<IoThreads> {
         serde_json::to_value(language::server_initialize_result(&init_param.capabilities))?;
     connection.initialize_finish(id, initialize_result)?;
 
-    main_loop(connection, init_param)?;
+    main_loop(connection, init_param, show_preview_ui)?;
 
     Ok(io_threads)
 }
 
-fn main_loop(connection: Connection, init_param: InitializeParams) -> Result<()> {
+fn main_loop(
+    connection: Connection,
+    init_param: InitializeParams,
+    show_preview_ui: bool,
```

Comment:
i'm thinking this could take the full CLI.    Also the fullscreen flag which use a env variable hack currently could be usefull,  as well as the style